plugins/outputs/microsoft_fabric/microsoft_fabric.go

```diff
@@ -0,0 +1,126 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package microsoft_fabric
+
+import (
+	_ "embed"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/plugins/outputs"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+type fabricOutput interface {
+	Init() error
+	Connect() error
+	Write(metrics []telegraf.Metric) error
+	Close() error
+}
+
+type MicrosoftFabric struct {
+	ConnectionString string          `toml:"connection_string"`
+	Log              telegraf.Logger `toml:"-"`
+	Timeout          config.Duration `toml:"timeout"`
+
+	eventhouse   *eventhouse
+	eventstream  *eventstream
+	activePlugin fabricOutput
```

Comment:
Why do you need all three of them?

Reply:
Idea is to maintain readability and clean design, with each type have its own functionality. Later it can be extended to have more type.